transfer/src/main/java/org/kinecosystem/transfer/repositories/EcosystemAppsRepository.kt

```diff
@@ -1,19 +1,21 @@
 package org.kinecosystem.transfer.repositories
 
+import android.content.Context
 import android.os.Handler
+import android.os.Looper
 import org.kinecosystem.transfer.model.*
 
 class EcosystemAppsRepository
 private constructor(private val currentPackage: String, private val local: EcosystemAppsLocalRepo,
                     private val remote: EcosystemAppsRemoteRepo, private val uiHandler: Handler) {
 
-
     companion object {
         private var instance: EcosystemAppsRepository? = null
 
-        fun getInstance(currentPackage: String, local: EcosystemAppsLocalRepo, remote: EcosystemAppsRemoteRepo, uiHandler: Handler): EcosystemAppsRepository {
+        fun getInstance(context: Context): EcosystemAppsRepository {
             if (instance == null) {
-                instance = EcosystemAppsRepository(currentPackage, local, remote, uiHandler)
+                instance = EcosystemAppsRepository(context.packageName, EcosystemAppsLocalRepo(context),
+                        EcosystemAppsRemoteRepo(), Handler(Looper.getMainLooper()))
```

Comment:
i thought the handler will be outside param so we can replace it on testing with some testing component and not hard coded mainlooper

Reply:
could create an additional getMockInstance method or in any case find a different way of managing this.
It doesn't make sense that each time we call get instance we create new instances of the repos that are not used!